File: src/main/java/com/market/MSA/requests/UpdateUserRequest.java
package com.market.MSA.requests;

import java.util.List;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class UpdateUserRequest {
  String fullname;

  String email;
  String phonenumber;
  String birthday;
  String password;
  String address;
  String googleid;

  List<Long> roles;
}

File: lib/feature/data/model/request/update_user_request.dart
class UpdateUserRequest {
  String fullName;
  String email;
  String phoneNumber;
  String birthday; // hoặc DateTime nếu bạn muốn dùng kiểu DateTime
  String password;
  String address;
  String googleId;
  List<int> roles;

  UpdateUserRequest({
    required this.fullName,
    required this.email,
    required this.phoneNumber,
    required this.birthday,
    required this.password,
    required this.address,
    required this.googleId,
    required this.roles,
  });

  factory UpdateUserRequest.fromJson(Map<String, dynamic> json) {
    return UpdateUserRequest(
      fullName: json['fullName'] ?? '',
      email: json['email'] ?? '',
      phoneNumber: json['phoneNumber'] ?? '',
      birthday: json['birthday'] ?? '',
      password: json['password'] ?? '',
      address: json['address'] ?? '',
      googleId: json['googleId'] ?? '',
      roles: List<int>.from(json['roles'] ?? []),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'fullName': fullName,
      'email': email,
      'phoneNumber': phoneNumber,
      'birthday': birthday,
      'password': password,
      'address': address,
      'googleId': googleId,
      'roles': roles,
    };
  }
}
